src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java

```diff
@@ -24,8 +25,16 @@
     @Override
     protected ActionResult execute() throws EntityDoesNotExistException {
         Assumption.assertNotNull(regkey);
```

Comment:
Hey, we could consider abstracting this away in Action.java. In the init() function we can check if there are ${amp}s in the complete url. If yes, break it up before coming to the individual Action level. 

Based on what you said, this problem could arise in other places and a common decoding placed at Action.java would help.